presto-spi/src/main/java/com/facebook/presto/spi/spiller/SpillStorageService.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.spiller;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+public interface SpillStorageService
+{
+    SpillFileHolder createTempFile(Path spillPath, String prefix, String suffix)
+            throws IOException;
+
+    void createDirectories(Path path)
+            throws IOException;
+
+    boolean canWrite(Path path);
+
+    double getAvailableSpacePercentage(Path path)
+            throws IOException;
+
+    void removeFilesQuitely(Path path, String fileGlob);
```

Comment:
I would recommend changing the interface to something more "high level".

For example

```
public interface StorageService {
  StorageHandle store(DataSink data);
  
  InputStream open(StorageHandle handle);
  
  void remove(StorageHandle handle);
}
```

Reply:
@arhimondr : 
with only `void remove(StorageHandle handle);`, how do we fulfill `cleanupOldSpillFiles` (which requires a glob?"

https://github.com/prestodb/presto/blob/83e4afc56e97764a276dc69e8789d53731425676/presto-main/src/main/java/com/facebook/presto/spiller/FileSingleStreamSpillerFactory.java#L131-L147


Also what's `StorageHandle` ?

Reply:
@arhimondr : Even in the future, I would assume this "storage service" is for temporary files (not a general purpose storage service) , thus some "best-effort" removal API seems to make sense. The actually API and name can evolve.

Reply:
Automatic cleanup should be the responsibility of the storage service, it doesn't have to be exposed to the engine.